src/api/roles/client/fitness.py

```
@@ -343,15 +388,106 @@ def _create_survey_response(survey, telemetry, completed_key: str, response_mode
+    if acc.client_id is None:
+        raise HTTPException(404, detail="Client profile not found")
+
+    cwp = db.get(ClientWorkoutPlan, payload.cwp_id)
+    if cwp is None or cwp.client_id != acc.client_id:
+        raise HTTPException(400, detail="Workout plan not found or not yours")
+
+    try:
+        from datetime import date as date_type
+        requested_date = datetime.strptime(payload.local_date, "%Y-%m-%d").date()
+    except ValueError:
+        raise HTTPException(400, detail="Invalid local_date format, expected YYYY-MM-DD")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)